Sources/SnapshotTesting/SnapshotTestCase.swift

```diff
@@ -1,126 +1,134 @@
+
+public func assertSnapshot<A: DefaultDiffable>(
+  matching snapshot: A,
+  named name: String? = nil,
+  record recording: Bool = false,
+  timeout: TimeInterval = 5,
+  file: StaticString = #file,
+  function: String = #function,
+  line: UInt = #line)
+  where A.A == A
+{
+  return assertSnapshot(
+    of: A.defaultStrategy,
+    matching: snapshot,
+    named: name,
+    record: recording,
+    timeout: timeout,
+    file: file,
+    function: function,
+    line: line
+  )
+}
 
-  public func assertSnapshot<A, B>(
-    of strategy: Strategy<A, B>,
-    matching value: A,
-    named name: String? = nil,
-    record recording: Bool = false,
-    timeout: TimeInterval = 5,
-    file: StaticString = #file,
-    function: String = #function,
-    line: UInt = #line
-    ) {
-
-    let recording = recording || self.record
-
-    do {
-      let snapshotDirectoryUrl: URL = {
-        let fileUrl = URL(fileURLWithPath: "\(file)")
-        let directoryUrl = fileUrl.deletingLastPathComponent()
-        return directoryUrl
-          .appendingPathComponent("__Snapshots__")
-          .appendingPathComponent(fileUrl.deletingPathExtension().lastPathComponent)
-      }()
-
-      let identifier: String
-      if let name = name {
-        identifier = name
-      } else {
-        identifier = String(counter)
-        counter += 1
+public func assertSnapshot<A, B>(
+  of strategy: Strategy<A, B>,
+  matching value: A,
+  named name: String? = nil,
+  record recording: Bool = false,
+  timeout: TimeInterval = 5,
+  file: StaticString = #file,
+  function: String = #function,
+  line: UInt = #line
+  ) {
+
+  let recording = recording || record
+
+  do {
+    let snapshotDirectoryUrl: URL = {
+      let fileUrl = URL(fileURLWithPath: "\(file)")
+      let directoryUrl = fileUrl.deletingLastPathComponent()
+      return directoryUrl
+        .appendingPathComponent("__Snapshots__")
+        .appendingPathComponent(fileUrl.deletingPathExtension().lastPathComponent)
+    }()
+
+    let identifier: String
+    if let name = name {
+      identifier = name
+    } else {
+      let counter = counterQueue.sync { () -> Int in
+        let key = snapshotDirectoryUrl.appendingPathComponent(function)
+        counterMap[key, default: 0] += 1
+        return counterMap[key]!
       }
+      identifier = String(counter)
+    }
 
-      let snapshotFileUrl = snapshotDirectoryUrl
-        .appendingPathComponent("\(function.dropLast(2)).\(identifier)")
-        .appendingPathExtension(strategy.pathExtension ?? "")
-      let fileManager = FileManager.default
-      try fileManager.createDirectory(at: snapshotDirectoryUrl, withIntermediateDirectories: true)
-
-      let tookSnapshot = self.expectation(description: "Took snapshot")
-      var optionalDiffable: B?
-      strategy.snapshotToDiffable(value).run { b in
-        optionalDiffable = b
-        tookSnapshot.fulfill()
-      }
-      #if os(Linux)
-      self.waitForExpectations(timeout: timeout)
```

Comment:
I'm hoping this doesn't fail. Linux got the additional APIs in recent snapshots...

Reply:
Dang, merged less than a week ago: https://github.com/apple/swift-corelibs-xctest/pull/228

Guess we can't merge and support Linux unless we lock Linux to a recent snapshot.